### PR TITLE
Implement websocket changes atop C# sdk changes

### PR DESCRIPTION
**Pull request by Centril**
## Requires SpacetimeDB PRs

This is the C# side of https://github.com/clockworklabs/SpacetimeDB/pull/1761.

### CODE REVIEW

**Comment by lcodes** [comment]
Oh good, most of the changes are to the clientAPI :)

File: src/SpacetimeDBClient.cs [hunk 1]
```diff
@@ -205,6 +203,97 @@ static DbValue Decode(IRemoteTableHandle table, EncodedValue value, out object?
         private static readonly Status Committed = new Status.Committed(default);
         private static readonly Status OutOfEnergy = new Status.OutOfEnergy(default);
 
+        private static ServerMessage DecompressDecodeMessage(byte[] bytes)
+        {
+            using var stream = new MemoryStream(bytes, 1, bytes.Length - 1);
+
+            switch (bytes[0])
```
Comment (lcodes) [inline src/SpacetimeDBClient.cs]: Is it possible for bytes to be an empty array? If so it'll fail here and in creating the memory stream

File: src/SpacetimeDBClient.cs [hunk 2]
```diff
@@ -205,6 +203,97 @@ static DbValue Decode(IRemoteTableHandle table, EncodedValue value, out object?
...
+            {
+                // Uncompressed.
+                case 0:
```
Comment (lcodes) [inline src/SpacetimeDBClient.cs]: Could create an enum for these, and cast the read byte to it

File: src/SpacetimeDBClient.cs [hunk 3]
```diff
@@ -205,6 +203,97 @@ static DbValue Decode(IRemoteTableHandle table, EncodedValue value, out object?
...
+                    }
+                // Brotli compressed.
+                case 1:
+                    {
+                        using var decompressedStream = new BrotliStream(stream, CompressionMode.Decompress);
+                        using var binaryReader = new BinaryReader(decompressedStream);
+                        return new ServerMessage.BSATN().Read(binaryReader);
+                    }
+                default:
+                    throw new InvalidOperationException();
```
Comment (lcodes) [inline src/SpacetimeDBClient.cs]: Can we have a reason argument to the exception? "Unknown compression type" or similar

Comment (lcodes) [reply src/SpacetimeDBClient.cs]: Same for the other exceptions

File: tests~/SnapshotTests.VerifyAllTablesParsed.verified.txt [hunk 4]
```diff
@@ -488,4 +486,4 @@
       Max: type=InitialSubscription
     }
   }
-}
\ No newline at end of file
+}
```
Comment (lcodes) [inline tests~/SnapshotTests.VerifyAllTablesParsed.verified.txt:393]: What's this diff?

File: src/SpacetimeDBClient.cs (hunk 1 above)
Comment (Centril) [reply src/SpacetimeDBClient.cs]: It's not possible. I'll add a comment.

File: tests~/SnapshotTests.VerifyAllTablesParsed.verified.txt (hunk 4 above)
Comment (Centril) [reply tests~/SnapshotTests.VerifyAllTablesParsed.verified.txt:393]: I'm not sure :/

**Review by lcodes — APPROVED** [review]

**Pull request merged**